File: src/2.add-two-numbers.py
#
# @lc app=leetcode id=2 lang=python3
#
# [2] Add Two Numbers
#

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
# @lc code=start
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        number1 = ""
        number2 = ""

        while l1 != None:
            number1 = str(l1.val) + number1
            l1 = l1.next

        while l2 != None:
            number2 = str(l2.val) + number2
            l2 = l2.next

        number_list = list(str(int(number1) + int(number2)))

        root = ListNode()
        current_node = root

        while len(number_list) > 0:
            current_number = number_list.pop()
            current_node.next = ListNode(current_number)
            current_node = current_node.next

        return root.next

# @lc code=end

File: Linked List/src/2.add-two-numbers.js
/*
 * @lc app=leetcode id=2 lang=javascript
 *
 * [2] Add Two Numbers
 */

// Definition for singly-linked list.
class ListNode {
  constructor(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? null : next)
  }
}
 
// @lc code=start
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
  let num1 = '';
  let num2 = '';

  let node1 = l1;
  while (node1 !== null) {
    num1 = node1.val + num1;
    node1 = node1.next;
  }

  let node2 = l2;
  while (node2 !== null) {
    num2 = node2.val + num2;
    node2 = node2.next;
  }

  const sum = BigInt(num1) + BigInt(num2);
  let sumNumbers = sum.toString().split('').reverse();

  const sumHead = new ListNode();
  let sumNode = sumHead;
  sumNumbers.forEach(val => {
    sumNode.next = new ListNode(val);
    sumNode = sumNode.next;
  });

  return sumHead.next;
};
// @lc code=end

